lib/backend/manager.go

```diff
@@ -60,15 +60,24 @@ func NewManager(configs []Config, auth AuthConfig, stats tally.Scope) (*Manager,
 		if len(config.Backend) != 1 {
 			return nil, fmt.Errorf("no backend or more than one backend configured")
 		}
-		var name string
+		var backendName string
 		var backendConfig interface{}
-		for name, backendConfig = range config.Backend { // Pull the only key/value out of map
+		for backendName, backendConfig = range config.Backend { // Pull the only key/value out of map
 		}
-		factory, err := getFactory(name)
+		factory, err := getFactory(backendName)
 		if err != nil {
 			return nil, fmt.Errorf("get backend client factory: %s", err)
 		}
-		c, err = factory.Create(backendConfig, auth[name], stats)
+		var backendAuthConfig interface{}
+		// Normally, when creating a backend client, we only need to pass its own auth credentials to its Create function.
+		// However, the shadow backend mode requires the auth configs of 2 backends (e.g. GCS and S3),
+		// thus we pass the whole AuthConfig (which includes the credentials of all used backends) to it. 
+		if backendName == "shadow" {
```

Comment:
This seems wrong for the manager to know of backend "shadow". The backendConfig could be anything. Could we instead depend on the backend to validate and parse the config keep the manager.go implementation the same?

Reply:
That's possible - instead of passing `auth[name]` to factory.Create, we would pass `auth` and each backend could parse the config as it needs. We'd need small changes to each backend's Create method.

Reply:
This makes a lot of sense. We should do it.